Mathlib/Logic/Equiv/Fin.lean

```diff
@@ -288,32 +288,36 @@ theorem finSuccEquivLast_symm_some (i : Fin n) :
 
 /-- Equivalence between `Π j : Fin (n + 1), α j` and `α i × Π j : Fin n, α (Fin.succAbove i j)`. -/
 @[simps (config := .asFn)]
-def Equiv.piFinSuccAboveEquiv (α : Fin (n + 1) → Type u) (i : Fin (n + 1)) :
+def Equiv.piFinSuccAbove (α : Fin (n + 1) → Type u) (i : Fin (n + 1)) :
     (∀ j, α j) ≃ α i × ∀ j, α (i.succAbove j) where
   toFun f := (f i, fun j => f (i.succAbove j))
   invFun f := i.insertNth f.1 f.2
   left_inv f := by simp [Fin.insertNth_eq_iff]
   right_inv f := by simp
-#align equiv.pi_fin_succ_above_equiv Equiv.piFinSuccAboveEquiv
-#align equiv.pi_fin_succ_above_equiv_apply Equiv.piFinSuccAboveEquiv_apply
-#align equiv.pi_fin_succ_above_equiv_symm_apply Equiv.piFinSuccAboveEquiv_symm_apply
+#align equiv.pi_fin_succ_above_equiv Equiv.piFinSuccAbove
+#align equiv.pi_fin_succ_above_equiv_apply Equiv.piFinSuccAbove_apply
+#align equiv.pi_fin_succ_above_equiv_symm_apply Equiv.piFinSuccAbove_symm_apply
 
 /-- Order isomorphism between `Π j : Fin (n + 1), α j` and
 `α i × Π j : Fin n, α (Fin.succAbove i j)`. -/
 def OrderIso.piFinSuccAboveIso (α : Fin (n + 1) → Type u) [∀ i, LE (α i)]
     (i : Fin (n + 1)) : (∀ j, α j) ≃o α i × ∀ j, α (i.succAbove j) where
-  toEquiv := Equiv.piFinSuccAboveEquiv α i
+  toEquiv := Equiv.piFinSuccAbove α i
   map_rel_iff' := Iff.symm i.forall_iff_succAbove
 #align order_iso.pi_fin_succ_above_iso OrderIso.piFinSuccAboveIso
 
 /-- Equivalence between `Fin (n + 1) → β` and `β × (Fin n → β)`. -/
 @[simps! (config := .asFn)]
 def Equiv.piFinSucc (n : ℕ) (β : Type u) : (Fin (n + 1) → β) ≃ β × (Fin n → β) :=
-  Equiv.piFinSuccAboveEquiv (fun _ => β) 0
+  Equiv.piFinSuccAbove (fun _ => β) 0
 #align equiv.pi_fin_succ Equiv.piFinSucc
 #align equiv.pi_fin_succ_apply Equiv.piFinSucc_apply
 #align equiv.pi_fin_succ_symm_apply Equiv.piFinSucc_symm_apply
 
+@[simps! (config := .asFn)]
+def Equiv.piFinCastSucc (n : ℕ) (β : Type u) : (Fin (n + 1) → β) ≃ β × (Fin n → β) :=
```

Comment:
It's tempting to flip the RHS to match the argument order for `snoc`:
```suggestion
def Equiv.piFinCastSucc (n : ℕ) (β : Type u) : (Fin (n + 1) → β) ≃ (Fin n → β) × β :=
```
(Also because this is visually extracting the last element)

Reply:
I am split on the matter because what do you do for `succAbove`? `(Fin a → β) × β × (Fin (n - a) → β)`?

Reply:
Eric makes a good point, but if I am the tiebreaker, I would prefer `β × (Fin n → β)`, because I think it might work better with #9040

Reply:
I'm happy with this.

Reply:
I personally don't see #9040 succeeding; I think the definitional properties of `snoc` and `cons` are different, and so directly translating proofs as we do with `+`/`*` is not going to work.

I agree that we should try to prove the snoc versions in terms of the cons versions where possible; but often this needs a lemma about reversal that needs both the cons and snoc lemmas to prove in the first place.

Reply:
Ok, well, I am basically ambivalent, I'm just going to commit this suggestion and maybe revisit it later.

Reply:
Ok I'm now reverting this, because as I now see, it breaks the build without changing the definition, and I like that the definition now is very minimal. If maintainers want to change it so be it.